### ISSUE TITLE
py2.7 dependency

### ISSUE THREAD

**Issue by ecervera** [RobInLabUJI/Webots-Docker#1]
@psychemedia Since I use the webots_ros controller, ROS is a dependency, and consequently is py2.7 too.

One option is to use a controller other than ROS: Webots now supports [both Python 3 and 2](https://www.cyberbotics.com/doc/guide/using-python).

**Comment by psychemedia** [comment]
Okay... that all means nothing to me TBH! ;-) (Not a ROS user, just trying to find ways of getting simple robot sims up for basic on-ramp edu / awareness purposes...)

Irrespective of the ROS dependency, would probably make sense to ensure that the Jupyter server envt run latest Jupyter packages, with the py2.7 ROS dependency installed in another py env. Notebooks can run whatever py version required as separate kernels?

**Comment by ecervera** [comment]
Yes, notebooks can have several kernels installed, namely Python 2 and Python 3, each with its own environment.
I will think about how to run Webots with a Python controller. I'm not an enthusiast of ROS, it was just the fastest option to begin with.

**Comment by psychemedia** [comment]
Not sure why that ? was at the end of my last sentence?! I guess point I was intending to make was the version of py that runs jupyter services is agnostic as to what language kernels the server runs, so makes sense to run recent Jupyter under py3 given pace of development of Jupyter environments.

**Comment by ecervera** [comment]
I have succeeded in getting rid of ROS and moving a robot with a Python 3 controller in Jupyter.
See the simple demo here: [https://github.com/RobInLabUJI/try-a-bot](https://github.com/RobInLabUJI/try-a-bot)
This is basically the Webots docker image with some additional files (next thing is to clean up and avoid replications).
